universal-application-tool-0.0.1/app/auth/AuthIdentityProviderName.java

```diff
@@ -0,0 +1,31 @@
+package auth;
+
+import java.util.Optional;
+
+/** Enum representing supported options for auth IDP for applicants and admins. */
+public enum AuthIdentityProviderName {
+  IDCS_APPLICANT("idcs"),
+  LOGIN_RADIUS_APPLICANT("login-radius"),
+  ADFS_ADMIN("adfs-admin"),
+  ;
+  private final String authIdentityProviderNameString;
+
+  AuthIdentityProviderName(String authIdentityProviderNameString) {
+    this.authIdentityProviderNameString = authIdentityProviderNameString;
+  }
+
+  /** Returns the enum associated with the provided string value. */
+  public static Optional<AuthIdentityProviderName> forString(String string) {
+    for (AuthIdentityProviderName authProvider : AuthIdentityProviderName.values()) {
```

Comment:
optional: Streams can rewrite this method as

return AuthIdentityProviderName.values().stream()
    .filter(provider -> provider.getString().equals(string))
    .findFirst()

Reply:
fixed